addon-test-support/-private/weight-test-modules.js

```diff
@@ -0,0 +1,50 @@
+const TEST_TYPE_WEIGHT = {eslint:1, unit:10, integration:20, acceptance:150};
+const WEIGHT_REGEX = /\/(eslint|unit|integration|acceptance)\//
+const DEFAULT_WEIGHT = 50;
+
+/**
+* Return the weight for a given module name, a file path to the module
+* Ember tests consist of Acceptance, Integration, Unit, and lint tests. In general, acceptance takes
+* longest time to execute, followed by integration and unit.
+* The weight assigned to a module corresponds to its test type execution speed, with slowest being the highest in weight.
+* If the test type is not identifiable from the modulePath, weight default to 50 (ordered after acceptance, but before integration)
+*
+* @param {string} modulePath File path to a module
+*/
+function getWeight(modulePath) {
+  const [, key] = WEIGHT_REGEX.exec(modulePath) || [];
+  if (typeof TEST_TYPE_WEIGHT[key] === 'number') {
+    return TEST_TYPE_WEIGHT[key] ;
+  } else {
+    return DEFAULT_WEIGHT;
+  }
+}
+
+/**
+ * Returns the list of modules sorted by its weight
+ *
+ * @export
+ * @param {Array<string>} modules
+ * @returns {Array<string>}
+ */
+export default function weightTestModules(modules) {
+  const groups = Object.create(null);
```

Comment:
Should we just be using a `Map` here?

Reply:
We can, do you see any difference between using a `Map` vs `Object.create(null);` in this case?

Reply:
`Map` is likely more clear and concise.  As long as `Map` is available where this code will be run, I think `Map` is a good bet.